app/scripts/controllers/user-storage/user-storage-controller.ts

```diff
@@ -27,15 +27,15 @@ export type UserStorageControllerState = {
   /**
    * Condition used by UI and to determine if we can use some of the User Storage methods.
    */
-  isProfileSyncingEnabled: boolean;
+  isProfileSyncingEnabled: boolean | null;
```

Comment:
Checked all references (JS and TS files) it seems okay to change this field to be nullable.

Reply:
We do not perform strict boolean checks in references, it is truthy checks (`isProfileSyncingEnabled && ...` or `isProfileSyncingEnabled || ...`)